### PR TITLE
Automatic benchmarking of various MOL discretizations across all of PDESystemLibrary

### PR DESCRIPTION
**Pull request by xtalax**
Automatically benchmark MethodOfLines.jl across the whole of PDESystemLibrary, generating WP diagrams for uniform grid, nonuniform Cebyshev grid, and with WENO scheme where the system is tagged with `"Advection"` - all with both `edge_align` and `center_align`.

I do not know the WP set syntax for `NonlinearSystem`, so time independent is skipped for now.

### CODE REVIEW

File: benchmarks/MethodOfLinesPDE/MOLxPDESystemLibrary.jl [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+            #         Dict(:alg => solver, :prob_choice => 3),
+            #         Dict(:alg => solver, :prob_choice => 4),]
+            #     names = ["Uniform Upwind, center_align", "Uniform Upwind, edge_align",
+            #              "Chebyshev Upwind, center_align", "Chebyshev Upwind, edge_align"];
+
+            #     wp = WorkPrecisionSet(probs1, abstols, reltols, setups; names=names,
+            #         save_everystep=false, appxsol = dummy_appxsol, maxiters=Int(1e5),
+            #         numruns=10, wrap=Val(false))
+            #     plot(wp, title=title)
+            # end
```
Comment (ChrisRackauckas) [inline benchmarks/MethodOfLinesPDE/MOLxPDESystemLibrary.jl:137]: ?

Comment (xtalax) [reply benchmarks/MethodOfLinesPDE/MOLxPDESystemLibrary.jl:137]: This was for the nonlinear systems, was unsure if `WPSet`  works for nonlinearsys - in any case analytic won't yet. Could I leave this skeleton in for when it does?

Comment (ChrisRackauckas) [reply benchmarks/MethodOfLinesPDE/MOLxPDESystemLibrary.jl:137]: I guess? It's not going to merge since it's failing anyways.

Comment (xtalax) [reply benchmarks/MethodOfLinesPDE/MOLxPDESystemLibrary.jl:137]: Yes, I need to change the argument signature for the analytic generation to include `p` in MOL. Doing this in the FunctionalScheme PR.

**Pull request merged**